modules/fs/fsw/src/cfe_fs_api.c

```diff
@@ -293,25 +293,16 @@ int32 CFE_FS_SetTimestamp(osal_id_t FileDes, CFE_TIME_SysTime_t NewTimestamp)
             CFE_FS_ByteSwapUint32(&OutTimestamp.Subseconds);
         }
 
-        Result = OS_write(FileDes, &OutTimestamp.Seconds, sizeof(OutTimestamp.Seconds));
+        Result = OS_write(FileDes, &OutTimestamp, sizeof(OutTimestamp));
 
         /* On a good write, the value returned will equal the number of bytes written */
-        if (Result == sizeof(OutTimestamp.Seconds))
+        if (Result == sizeof(OutTimestamp))
         {
-            Result = OS_write(FileDes, &OutTimestamp.Subseconds, sizeof(OutTimestamp.Subseconds));
-
-            if (Result == sizeof(OutTimestamp.Subseconds))
-            {
-                Result = OS_SUCCESS;
-            }
-            else
-            {
-                CFE_ES_WriteToSysLog("CFE_FS:SetTime-Failed to write Seconds (Status=0x%08X)\n", (unsigned int)Result);
-            }
+            Result = OS_SUCCESS;
         }
         else
         {
-            CFE_ES_WriteToSysLog("CFE_FS:SetTime-Failed to write Seconds (Status=0x%08X)\n", (unsigned int)Result);
+            CFE_ES_WriteToSysLog("%s(): Failed to write timestamp (Status=0x%08X)\n", __func__, (unsigned int)Result);
```

Comment:
To be pendantic - this should overwrite the "Result" to something that is known to be an error.  I would suggest `CFE_STATUS_EXTERNAL_RESOURCE_FAIL`.

Note that the "write" call above could return something that is positive but less than `sizeof(OutTimestamp)` or it could even return 0, which would be interpreted as success, even though it was not successful.

Reply:
Fixed